### PR TITLE
add discovery functional tests when discovery is forced or disabled

### PR DESCRIPTION
**Pull request by fbourigault**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | related to #183 
| Documentation   | n/a
| License         | MIT

This add more tests related to discovery and will help to get #183 refactoring safe.

### CODE REVIEW

File: Tests/Resources/app/AppKernel.php [hunk 1]
```diff
@@ -28,9 +28,10 @@ public function registerBundles()
      */
     public function registerContainerConfiguration(LoaderInterface $loader)
     {
-        $this->isDebug()
-            ? $loader->load(__DIR__.'/config/config_debug.yml')
-            : $loader->load(__DIR__.'/config/config.yml');
+        $loader->load(__DIR__.'/config/config_'.$this->getEnvironment().'.yml');
+        if ($this->isDebug()) {
+            $loader->load(__DIR__.'/config/config_debug.yml');
+        }
```
Comment (Nyholm) [inline Tests/Resources/app/AppKernel.php]: else {
//load config/config.yml
}

Comment (fbourigault) [inline Tests/Resources/app/AppKernel.php]: `config.yml` have been renamed to `config_test.yml` to always have a configuration file matching the current environment.

File: Tests/Functional/DiscoveredClientsTest.php [hunk 2]
```diff
@@ -71,9 +71,32 @@ public function testDiscoveredAsyncClientWithProfilingEnabled()
...
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_async'));
+        $this->assertFalse($container->has('httplug.strategy'));
+    }
+
+    public function testForcedDiscovery()
+    {
+        $container = $this->getContainer(true, 'discovery_forced');
+
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_client'));
+        $this->assertFalse($container->has('httplug.auto_discovery.auto_discovered_async'));
```
Comment (Nyholm) [inline Tests/Functional/DiscoveredClientsTest.php]: Should these be false on both `testDisabledDiscovery` and `testForcedDiscovery`?

Comment (fbourigault) [inline Tests/Functional/DiscoveredClientsTest.php]: Yes, when forced, we use the service id provided by the user. IMHO, those services shouldn't be public as we use them as `httplug.discovery` arguments.

**Pull request merged**